File: Python/Functional-2/no_long.py

# given a list of strings, return a list of the strings, omitting any string length 4 or more
def no_long(strings):
    new_list = []

    for x in strings:
        if len(x) < 4:
            new_list.append(x)

    return new_list

print(no_long(['this', 'not', 'too', 'long']))
print(no_long(['a', 'bbb', 'cccc']))
print(no_long(['cccc', 'cccc', 'cccc']))

File: JavaScript/Functional-2/noLong.js

// given a list of strings, return a list of the strings, omitting any string length 4 or more
const noLong = (strings) => {
    const newList = strings.filter(n => n.length < 4);

    return newList;
};

console.log(noLong(['this', 'not', 'too', 'long']));
console.log(noLong(['a', 'bbb', 'cccc']));
console.log(noLong(['cccc', 'cccc', 'cccc']));
